packages/@aws-cdk/aws-codepipeline-api/lib/build-action.ts

```diff
@@ -48,8 +48,6 @@ export abstract class BuildAction extends Action {
     });
 
     this.addInputArtifact(props.inputArtifact);
-    if (props.artifactName) {
-      this.artifact = this.addOutputArtifact(props.artifactName);
-    }
+    this.artifact = this.addOutputArtifact(props.artifactName);
```

Comment:
This is a change in behavior. This means that now `BuildAction` will always have an output artifact defined for it. Could be totally fine, but what happens if the CodeBuild project doesn't specify any artifacts in `buildspec.yaml` but you still define an output artifact?

Reply:
Good call out. I'll check what happens.

Reply:
Confirmed it doesn't cause any problems - there simply is no output artifact produced from the CodeBuild Project in this case, but the build Action itself succeeds without issues.